Sources/ClientRuntime/Networking/Http/URLSession/FoundationStreamBridge.swift

```diff
@@ -182,20 +157,22 @@ class FoundationStreamBridge: NSObject, StreamDelegate {
         }
     }
 
-    private class WriteToOutputStreamResult: NSObject {
-        var data = Data()
-        var error: Error?
-    }
-
     /// Write the passed data to the output stream, using the reserved thread.
     private func writeToOutputStream(data: Data) async throws {
         try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
-            Self.queue.async {
-                let result = WriteToOutputStreamResult()
-                result.data = data
-                let selector = #selector(self.writeToOutputStreamOnThread)
-                self.perform(selector, on: Self.thread, with: result, waitUntilDone: true)
-                if let error = result.error {
+            queue.async { [self] in
+                guard !buffer.isEmpty || !data.isEmpty else { continuation.resume(); return }
```

Comment:
When I looking at this code during checksums, I was wondering should we check if `outputStream.streamStatus == .open` ?

Reply:
The stream will error below if we write to it while not open, I think that is sufficient.  

(This method is only called as a result of a stream delegate callback, and that callback only happens when the stream is open, and the `writeToOutput()` method above enforces concurrent access with a check that the stream has not already closed, so as a practical matter, a write will never happen when the stream is closed.)